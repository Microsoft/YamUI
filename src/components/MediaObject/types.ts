/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum MediaObjectSize {
  XLARGE = 'xLarge',
  LARGE = 'large',
  MEDIUM = 'medium',
  SMALL = 'small',
  XSMALL = 'xSmall',
}
