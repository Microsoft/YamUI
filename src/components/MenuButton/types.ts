/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { ContextualMenuItemType } from 'office-ui-fabric-react/lib/ContextualMenu';

export enum MenuItemType {
  Normal = ContextualMenuItemType.Normal,
  Header = ContextualMenuItemType.Header,
  Divider = ContextualMenuItemType.Divider,
}
