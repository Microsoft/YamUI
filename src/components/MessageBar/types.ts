/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum MessageBarType {
  INFO = 'info',
  WARNING = 'warning',
  ERROR = 'error',
}
