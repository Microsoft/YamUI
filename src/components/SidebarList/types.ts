/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { BaseIllustration } from '../Illustration';

export type SidebarListSize = 'small' | 'medium' | 'large';

export interface BaseSidebarListItem {
  /**
   * A unique key for the item.
   */
  key: string;

  /**
   * The visible title text.
   */
  title: string;

  /**
   * Metadata text displayed under the title, not displayed at small size.
   */
  description?: string;
}
export interface SidebarListIllustrationItem extends BaseSidebarListItem {
  /**
   * A YamUI Illustration component class. You are responsible for importing and providing the correct size.
   */
  illustration: typeof BaseIllustration;
}
export interface SidebarListImageItem extends BaseSidebarListItem {
  /**
   * Custom image JSX. It should fit the list item's pixel width at the given size, or fill 100% of its container width.
   */
  imageContent: React.ReactNode;
}
export type SidebarListItemWithGraphic = SidebarListIllustrationItem | SidebarListImageItem;

export type SidebarListClickableItem = SidebarListItemWithGraphic & {
  /**
   * Callback to be triggered when the item is clicked/invoked.
   */
  onClick(key: BaseSidebarListItem['key']): void;
};
export type SidebarListLinkItem = SidebarListItemWithGraphic & {
  /**
   * The URL this item links to.
   */
  href: string;

  /**
   * If true, opens the link securely in a new window.
   */
  newWindow?: boolean;
};
export type SidebarListItem = SidebarListClickableItem | SidebarListLinkItem;
