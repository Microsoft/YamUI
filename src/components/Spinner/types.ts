/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum SpinnerColor {
  /**
   * Light theme.
   */
  LIGHT = 'light',
  /**
   * Light theme spinner color with gray text.
   */
  METADATA = 'metadata',
  /**
   * Dark theme.
   */
  DARK = 'dark',
}
export enum SpinnerSize {
  LARGE = 'large',
  MEDIUM = 'medium',
  SMALL = 'small',
  XSMALL = 'xSmall',
}
