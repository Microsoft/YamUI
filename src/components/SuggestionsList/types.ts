/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */

import { AvatarProps } from '../Avatar';

export enum SuggestionsListItemTemplate {
  TEXT = 'text',
  AVATAR_ROUND = 'mediaRound',
  MEDIA_SOFT = 'mediaSoft',
  MEDIA_SQUARE = 'mediaSquare',
}

export type CustomizableAvatarProps = 'badgeContent' | 'badgeDescription';

export interface BaseSuggestionItem {
  id: string | number;
  name: string;
}
export interface TextSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.TEXT;
}
export interface MediaSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.MEDIA_SOFT | SuggestionsListItemTemplate.MEDIA_SQUARE;
  imageUrl?: string;
  description?: string;
}
export interface AvatarSuggestionItem extends BaseSuggestionItem {
  template: SuggestionsListItemTemplate.AVATAR_ROUND;
  imageUrl?: string;
  avatarProps?: Pick<AvatarProps, CustomizableAvatarProps>;
  description?: string;
}
export type SuggestionItem = TextSuggestionItem | MediaSuggestionItem | AvatarSuggestionItem;
