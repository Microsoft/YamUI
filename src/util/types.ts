/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
