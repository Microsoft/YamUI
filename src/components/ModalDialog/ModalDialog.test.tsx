/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ModalDialog, ModalDialogProps } from './index';
import { ModalDialogSize } from './types';

describe('<ModalDialog />', () => {
  let component: ShallowWrapper<ModalDialogProps>;
  let onDismiss: jest.Mock;

  describe('with default options', () => {
    beforeEach(() => {
      onDismiss = jest.fn();
      component = shallow(
        <ModalDialog
          title={'Title'}
          isOpen={false}
          onDismiss={onDismiss}
          size={ModalDialogSize.MEDIUM}
          closeAriaLabel={'close'}
        />,
      );
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      onDismiss = jest.fn();
      component = shallow(
        <ModalDialog
          title={'Title'}
          className="TEST_CLASSNAME"
          isOpen={false}
          onDismiss={onDismiss}
          size={ModalDialogSize.MEDIUM}
          closeAriaLabel={'close'}
        />,
      );
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the title is hidden', () => {
    beforeEach(() => {
      onDismiss = jest.fn();
      component = shallow(
        <ModalDialog
          title={'Title'}
          className="TEST_CLASSNAME"
          isOpen={false}
          onDismiss={onDismiss}
          size={ModalDialogSize.MEDIUM}
          closeAriaLabel={'close'}
          hideTitle={true}
        />,
      );
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
