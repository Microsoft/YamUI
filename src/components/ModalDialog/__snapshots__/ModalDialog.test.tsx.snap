// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalDialog /> when the title is hidden matches its snapshot 1`] = `
<WithResponsiveMode
  className=""
  containerClassName=
      y-modalDialog
      TEST_CLASSNAME
      {
        width: 580px;
      }
  isBlocking={false}
  isDarkOverlay={true}
  isOpen={false}
  onDismiss={[MockFunction]}
  titleAriaId="y-modalDialog--title"
>
  <CustomizedBlock
    horizontalPadding="xLarge"
    verticalPadding="medium"
  >
    <ScreenreaderText>
      <div
        id="y-modalDialog--title"
      >
        <h1>
          Title
        </h1>
      </div>
    </ScreenreaderText>
  </CustomizedBlock>
</WithResponsiveMode>
`;

exports[`<ModalDialog /> with additional className matches its snapshot 1`] = `
<WithResponsiveMode
  className=""
  containerClassName=
      y-modalDialog
      TEST_CLASSNAME
      {
        width: 580px;
      }
  isBlocking={false}
  isDarkOverlay={true}
  isOpen={false}
  onDismiss={[MockFunction]}
  titleAriaId="y-modalDialog--title"
>
  <CustomizedBlock
    horizontalPadding="xLarge"
    verticalPadding="medium"
  >
    <FixedGridRow
      bottomSpacing="xSmall"
    >
      <FixedGridColumn>
        <div
          id="y-modalDialog--title"
        >
          <Heading
            level="1"
            size="5"
          >
            Title
          </Heading>
        </div>
      </FixedGridColumn>
      <FixedGridColumn
        fixed={true}
      >
        <Clickable
          ariaLabel="close"
          block={true}
          onClick={[MockFunction]}
          unstyled={true}
        >
          <Cancel
            block={true}
            size="12"
          />
        </Clickable>
      </FixedGridColumn>
    </FixedGridRow>
  </CustomizedBlock>
</WithResponsiveMode>
`;

exports[`<ModalDialog /> with default options matches its snapshot 1`] = `
<WithResponsiveMode
  className=""
  containerClassName=
      y-modalDialog
      {
        width: 580px;
      }
  isBlocking={false}
  isDarkOverlay={true}
  isOpen={false}
  onDismiss={[MockFunction]}
  titleAriaId="y-modalDialog--title"
>
  <CustomizedBlock
    horizontalPadding="xLarge"
    verticalPadding="medium"
  >
    <FixedGridRow
      bottomSpacing="xSmall"
    >
      <FixedGridColumn>
        <div
          id="y-modalDialog--title"
        >
          <Heading
            level="1"
            size="5"
          >
            Title
          </Heading>
        </div>
      </FixedGridColumn>
      <FixedGridColumn
        fixed={true}
      >
        <Clickable
          ariaLabel="close"
          block={true}
          onClick={[MockFunction]}
          unstyled={true}
        >
          <Cancel
            block={true}
            size="12"
          />
        </Clickable>
      </FixedGridColumn>
    </FixedGridRow>
  </CustomizedBlock>
</WithResponsiveMode>
`;
